tls/s2n_resume.h

```diff
@@ -99,3 +104,13 @@ extern int s2n_store_to_cache(struct s2n_connection *conn);
  * once we release the session resumption API. */
 int s2n_config_set_initial_ticket_count(struct s2n_config *config, uint8_t num);
 int s2n_connection_add_new_tickets_to_send(struct s2n_connection *conn, uint8_t num);
+
+struct s2n_session_ticket {
+    struct s2n_blob ticket_data;
+    uint32_t session_lifetime;
+};
+
+typedef int (*s2n_session_ticket_cb)(struct s2n_connection *conn, struct s2n_session_ticket *ticket);
+int s2n_config_set_session_ticket_callback(struct s2n_config *config, s2n_session_ticket_cb cb);
+int s2n_session_ticket_get_data(struct s2n_session_ticket *ticket, uint8_t **data, size_t *data_len);
+int s2n_session_ticket_get_lifetime(struct s2n_session_ticket *ticket, size_t *session_lifetime);
```

Comment:
Since these will eventually be public can you write documentation about them inline? See [`s2n_config_set_key_log_cb`](https://github.com/aws/s2n-tls/blob/463abf9e62dfcfbf9dc270de3dc80cf4c740db92/api/s2n.h#L511-L527) for an example: 

https://github.com/aws/s2n-tls/blob/463abf9e62dfcfbf9dc270de3dc80cf4c740db92/api/s2n.h#L511-L527

Each one should ideally have a `# Safety` section describing exactly what the caller needs to do to ensure the function is memory safe. For example: the `s2n_session_ticket_get_data` function that we're adding hands out our internal pointer to the caller. We need to detail _exactly_ what a caller can do with that pointer.

Reply:
Idk, we haven't been doing that so far. I think the plan is to write the documentation when we move them to s2n.h, rather than keeping it up to date during development.

Reply:
I'm going to leave the documentation out for this pr, but you're right Cameron, we do need to explicitly document this function.